tutorials/roofit/rf317_global_observables.C

```diff
@@ -0,0 +1,169 @@
+/// \file
+/// \ingroup tutorial_roofit
+/// \notebook -js
+/// This tutorial explains the concept of global observables in RooFit, and
+/// showcases how their values can be stored either in the model or in the
+/// dataset.
+///
+/// \macro_code
+///
+/// \date January 2022
+/// \author Jonas Rembser
+
+#include <RooArgSet.h>
+#include <RooConstVar.h>
+#include <RooDataSet.h>
+#include <RooFitResult.h>
+#include <RooGaussian.h>
+#include <RooProdPdf.h>
+#include <RooRealVar.h>
+
+// Helper function to call RooAbsPdf::fitTo.
+template <class... Args_t>
+std::unique_ptr<RooFitResult> doFit(RooAbsPdf &model, RooAbsData &data, Args_t &&...args)
+{
+   using namespace RooFit;
+
+   // we clone the model to not change the parameters by fitting
+   std::unique_ptr<RooAbsPdf> modelClone{static_cast<RooAbsPdf *>(model.cloneTree())};
```

Comment:
Why such an expensive operation? I think it's better to use the model and reset the parameters if desired. Something like this (+ the things you need to change to make it compile ... 🙂 )
```c++
RooArgSet modelParameters{ model.getParameters() }; // or whatever is necessary
RooArgSet origParameters = modelParameters.snapshot();

model.fitTo(...);
modelParameters = origParameters;
```